swap/src/protocol/alice/swap.rs

```diff
@@ -81,10 +64,31 @@ pub fn is_encsig_learned(state: &AliceState) -> bool {
     )
 }
 
+pub async fn run(swap: alice::Swap) -> Result<AliceState> {
+    run_until(swap, is_complete).await
+}
+
+pub async fn run_until(
+    swap: alice::Swap,
+    is_target_state: fn(&AliceState) -> bool,
+) -> Result<AliceState> {
+    do_run_until(
+        swap.state,
+        is_target_state,
+        swap.event_loop_handle,
+        swap.bitcoin_wallet,
+        swap.monero_wallet,
+        swap.config,
+        swap.swap_id,
+        swap.db,
+    )
+    .await
+}
+
 // State machine driver for swap execution
 #[async_recursion]
 #[allow(clippy::too_many_arguments)]
-pub async fn run_until(
+async fn do_run_until(
```

Comment:
NITPICK: why do_run_until instead of run_until? It seems like an unnecessary addition of a word

Reply:
Because I added another `run_until` function that is exposed and takes the `Swap` as an argument. That way we have the internal function and a wrapper function that is exposed. I was not able to give it the same name because Rust does not support function overloading where the function would have the same name but different parameters :)

Reply:
Note: I did try to achieve this by using the object, but ran into lifetime issues (notably mutably borrowing the event loop handle) that could not be resolved easily. I am not sure this can be easily solved due to the recursive function calls.